crates/rattler_solve/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+#![deny(missing_docs)]
+
+//! `rattler_solve` is a crate that provides functionality to solve Conda environments. It currently
+//! exposes the functionality through the [`SolverProblem::solve`] function.
+
+mod libsolv;
+mod package_operation;
+
+pub use package_operation::{PackageIdentifier, PackageOperation, PackageOperationKind};
+
+use rattler_conda_types::{MatchSpec, RepoData};
+
+/// Represents an error when solving the dependencies for a given environment
+#[derive(thiserror::Error, Debug)]
+pub enum SolveError {
+    /// There is no set of dependencies that satisfies the requirements
+    #[error("unsolvable")]
+    Unsolvable,
+
+    /// An error occurred when trying to load the channel and platform's `repodata.json`
+    #[error("error adding repodata: {0}")]
+    ErrorAddingRepodata(#[source] anyhow::Error),
+
+    /// An error occurred when trying to load information about installed packages to the solver
+    #[error("error adding installed packages: {0}")]
+    ErrorAddingInstalledPackages(#[source] anyhow::Error),
+
+    /// The solver backend returned operations that have no known mapping to [`PackageOperationKind`]
+    #[error("unsupported operations")]
+    UnsupportedOperations,
```

Comment:
This error is not very informative to the user. Which operation could not be mapped for instance?

Reply:
The idea is to keep the errors backend-agnostic, so basically this error is saying: _your libsolv backend is buggy, because it is returning operations that you don't know how to interpret_. If we want to provide a bit more context, I can think of the following alternatives:

1. Add logging on the libsolv backend side, so the user gets to see something at stderr about unsupported operations.
2. Modify the `UnsupportedOperations` variant to include a `Vec<String>` which can be populated by the solver backend to provide a user-friendly representation of the operations that could not be translated).

I think I'd prefer 2. Do you agree? Or would you suggest a third approach?

Reply:
I agree completely :)